wifiphisher/common/phishinghttp.py

```diff
@@ -157,6 +162,17 @@ def post(self):
         file_path = template_directory + render_file
         self.render(file_path, **template.get_context())
 
+        """
+        finds victim object that corresponds to ip addres and
+        associate to its hostname by comparing the url to the initial requests file and update its victims_dic
+        """
+        ip_addr = self.request.remote_ip.strip()
+        url = self.request.full_url().strip()
+        victims_instance = victim.Victims.getInstance()
+        for key in victims_instance.victims_dic.keys():
+            if ip_addr == victims_instance.victims_dic[key].ip_address:
+                victim_object = victims_instance.victims_dic[key]
+                victim_object.associate_victim_ip_to_hostname(ip_addr, url)
```

Comment:
This snippet here should be a method in `Victim` class. Something like `update_victim_os`.

Reply:
Also, note that this is association between victim and OS (and not hostname).